sourcetracker/_plot.py

```diff
@@ -10,16 +10,306 @@
+        returns
+        --------
+        none
+        outputs a heatmap visualization in a PNG
+        """
+        prop = self.mpm
+        if not unknowns:
+            prop = prop.drop(['Unknown'], axis=1)
+            prop = prop.div(prop.sum(axis=1), axis=0)
+        fig, ax = plt.subplots(figsize=((prop.shape[1] * 3 / 4)+4,
+                                        (prop.shape[0] * 3 / 4)+4))
+        sns.heatmap(prop, vmin=0.0, vmax=vmax, cmap=self.color,
+                    annot=annot, linewidths=.5, ax=ax)
+        ax.set_xlabel(xlabel)
+        ax.set_ylabel(ylabel)
+        ax.set_title(self.title)
+        plt.xticks(rotation=45, ha='right')
+        if unknowns:
+            plt.savefig(os.path.join(self.file,
+                                     self.out_name + "_heatmap.png"))
+        else:
+            plt.savefig(os.path.join(self.file,
+                                     self.out_name + "_heatmap_nounknown.png"))
+
+    def ST_paired_heatmap(self, normalized=False, unknowns=True,
+                          transpose=False, annot=True, ylabel='Sinks',
+                          heat_ratio=0.08):
+        """
+        Parameters
+        ----------
+        normalized bool
+            normalize each column to equal to 1 to represent the likelihood
+        unknowns bool
+            removes unknown column
+        transpose bool
+            transpose
+        annot bool
+            Defines visual proportions in plot
+        xlabel string
+            x label
+        ylabel string
+            y label
+        heat ratio float
+            ratio of example bar as compared to main columns.
+            Should be much thinner than main columns
+        returns
+        --------
+        none
+        outputs a heatmap visualization in a PNG defined by each
+        individual column
+
+        Any analysis should be done using a bin(n,x)
+
+        Color examples:
+        "viridis" "icefire" "vlag" "Spectral" "mako" "magma"
+        "coolwarm" "rocket" "flare" "crest"
+        "_r" reverses all of these
+
+        vmax and min will show the maximum and minimum for the
+        heat map settings. vmax=max is default and what we use here.
+        vmin=min is also what we use here but the standard version
+        will use vmin=0 in order to show the minimum possible
+        and vmax=1.
+        The reason I do not in this case is that these ranges
+        are not particularly helpful to distinguishing the
+        successful matches to each other.
+
+        Paired heatmap should be used for assessing overlap in
+        paired individual microbiomes or metabolomic overlap.
+        Some examples of this may include organ donors and
+        recipients, matching animal fecal samples to track
+        movement patterns, overlap in mouse microbiomes
+        caged together, and individual identification
+        given a range of possible known samples.
+        """
```

Comment:
Because of the split in proportions the issue is the visualization will be not very readable. I encountered this problem when I tried to make these heatmaps with individual pairings. The numbers are still there yes but there doesn't seem to be a good way to visualize this kind of question without separating each column into it's own heatmap. TLDR: The data is not the issue but the difference between purple and dark blue is a lot less stark than yellow to purple.